JASP-Engine/JASP/R/regressionlinearbayesian.R

```diff
@@ -611,1138 +384,1065 @@ RegressionLinearBayesian <- function (
-            ggplot2::geom_point(size = 4) + 
-            ggplot2::geom_errorbar(, width = 0.2) + 
-            ggplot2::scale_x_discrete(name = "") + 
-            ggplot2::scale_y_continuous(name = expression(beta), breaks = yBreaks, limits = range(yBreaks))
-        plotObj <- JASPgraphs::themeJasp(g) + 
-            ggplot2::theme(
-                axis.title.y = ggplot2::element_text(angle = 0, vjust = .5, size = 20)
-            )
+.basregFillPlotModelComplexity <- function(basregModel, jaspResults, options) {
+  
+  p <- try({
+    
+    logmarg = basregModel$logmarg
+    dim = basregModel$size
+    
+    dfPoints <- data.frame(
+      x = dim,
+      y = logmarg
+    )
+    
+    # gonna assume here that dim (the number of parameters) is always an integer
+    xBreaks <- unique(round(pretty(dim)))
+    yBreaks <- JASPgraphs::getPrettyAxisBreaks(range(logmarg), eps.correct = 2)
+    g <- JASPgraphs::drawPoints(dat = dfPoints, size = 4) +
+      ggplot2::scale_y_continuous(name = "Log(P(data|M))", breaks = yBreaks, limits = range(yBreaks)) +
+      ggplot2::scale_x_continuous(name = "Model Dimension", breaks = xBreaks)
+    JASPgraphs::themeJasp(g)
+    
+  })
+  
+  if (isTryError(p)) {
+    errorMessage <- paste("Plotting not possible:", .extractErrorMessage(p))
+    jaspResults[["modelComplexityPlot"]]$error <- "badData" #FIXME: this should go
+    jaspResults[["modelComplexityPlot"]]$errorMessage <- errorMessage
+  } else {
+    jaspResults[["modelComplexityPlot"]]$plotObject <- p
+  }
+  
+  return()
+}
 
-        w <- 530
-        h <- 400
-        content <- .writeImage(width = w, height = h, plot = plotObj)
-        plot <- list(
-            title = title,
-            width = w,
-            height = h,
-            convertible = TRUE,
-            status = "complete",
-            obj = content[["obj"]],
-            data = content[["png"]]
-        )
+.basregPlotInclusionProbabilities <- function(basregModel, jaspResults, options, status) {
+  if (!is.null(jaspResults[["inclusionProbabilitiesPlot"]]) ||
+      !options$plotInclusionProbabilities)
+    return()
+  
+  inclusionProbabilitiesPlot <- createJaspPlot(title = "Inclusion Probabilities", width = 700, height = 400)
+  jaspResults[["inclusionProbabilitiesPlot"]] <- inclusionProbabilitiesPlot
+  
+  inclusionProbabilitiesPlot$copyDependenciesFromJaspObject(jaspResults[["stateModel"]])
+  inclusionProbabilitiesPlot$dependOnOptions("plotInclusionProbabilities")
+  
+  if (!status$ready) {
+    return()
+  } else if (status$error) {
+    inclusionProbabilitiesPlot$error <- "badData"
+    return()
+  }
+  
+  .basregFillPlotInclusionProbabilities(basregModel, jaspResults, options)
+  
+  return()
+}
 
-    } else {
-        plot <- .makeEmptyPlot.basReg(title = title, status = status)
-    }
+.basregFillPlotInclusionProbabilities <- function(basregModel, jaspResults, options) {
+  
+  p <- try({
```

Comment:
In this and the other `try({...})` blocks the first 20 lines aren't really prone to erroring. Perhaps it's more clear to only put things in `try({...})` that can give an error?